cv32e40s/bsp/bsp.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 Silicon Laboratories Inc.
+//
+// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+//
+// Licensed under the Solderpad Hardware License v 2.1 (the "License"); you
+// may not use this file except in compliance with the License, or, at your
+// option, the Apache License version 2.0.
+//
+// You may obtain a copy of the License at
+// https://solderpad.org/licenses/SHL-2.1/
+//
+// Unless required by applicable law or agreed to in writing, any work
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+typedef union {
+  struct {
+    volatile uint32_t  load     : 1;
+    volatile uint32_t  store    : 1;
+    volatile uint32_t  execute  : 1;
+    volatile uint32_t  u        : 1;
+    volatile uint32_t  s        : 1;
+    volatile uint32_t  res_5_5  : 1;
+    volatile uint32_t  m        : 1;
+    volatile uint32_t  match    : 4;
+    volatile uint32_t  chain    : 1;
+    volatile uint32_t  action   : 4;
+    volatile uint32_t  size     : 4;
+    volatile uint32_t  timing   : 1;
+    volatile uint32_t  select   : 1;
+    volatile uint32_t  hit      : 1;
+    volatile uint32_t  vu       : 1;
+    volatile uint32_t  vs       : 1;
+    volatile uint32_t  res_26_25: 2;
+    volatile uint32_t  dmode    : 1;
+    volatile uint32_t  type     : 4;
+  } __attribute__((packed)) volatile fields;
+  volatile uint32_t raw;
+} __attribute__((packed)) mcontrol6_t;
```

Comment:
should we perhaps put this in some other file in the bsp folder rather than including it directly in bsp.h; otherwise this will get very cluttered very fast (previously these headers have generally been added to the tests directly)

Reply:
nevermind the first comment - I realized this is a new file